packages/react-components/react-menu/src/components/MenuPopover/useMenuPopover.ts

```diff
@@ -60,51 +60,38 @@ export const useMenuPopover_unstable = (props: MenuPopoverProps, ref: React.Ref<
   const inline = useMenuContext_unstable(context => context.inline) ?? false;
   const mountNode = useMenuContext_unstable(context => context.mountNode);
 
-  const rootProps = getNativeElementProps('div', {
-    role: 'presentation',
-    ...restoreFocusSourceAttributes,
-    ...props,
-    ref: useMergedRefs(ref, popoverRef, mouseOverListenerCallbackRef),
-  });
-
+  const rootProps = slot.always(
+    getNativeElementProps('div', {
+      role: 'presentation',
+      ...restoreFocusSourceAttributes,
+      ...props,
+      ref: useMergedRefs(ref, popoverRef, mouseOverListenerCallbackRef),
+    }),
+    { elementType: 'div' },
+  );
   const { onMouseEnter: onMouseEnterOriginal, onKeyDown: onKeyDownOriginal } = rootProps;
-
   rootProps.onMouseEnter = useEventCallback((event: React.MouseEvent<HTMLElement>) => {
     if (openOnHover) {
       setOpen(event, { open: true, keyboard: false, type: 'menuPopoverMouseEnter', event });
     }
-
     onMouseEnterOriginal?.(event);
   });
-
   rootProps.onKeyDown = useEventCallback((event: React.KeyboardEvent<HTMLElement>) => {
     const key = event.key;
-
     if (key === Escape || (isSubmenu && key === CloseArrowKey)) {
       if (open && popoverRef.current?.contains(event.target as HTMLElement)) {
-        setOpen(event, { open: false, keyboard: true, type: 'menuPopoverKeyDown', event });
-        // stop propagation to avoid conflicting with other elements that listen for `Escape`
+        setOpen(event, { open: false, keyboard: true, type: 'menuPopoverKeyDown', event }); // stop propagation to avoid conflicting with other elements that listen for `Escape`
```

Comment:
```suggestion
        setOpen(event, { open: false, keyboard: true, type: 'menuPopoverKeyDown', event });
        // stop propagation to avoid conflicting with other elements that listen for `Escape`
```

Probably should stay as it was